pyanaconda/ui/tui/tuiobject.py

```diff
@@ -55,3 +86,223 @@ def has_help(self):
+        :type title: str
+
+        :param value: Actual value of the title. User will see what is set right now.
+        :type value: str
+
+        :param conditions: Optional acceptance conditions. If condition is not valid the
+                          `wrong_input_message` is printed and user must set correct value.
+        :type conditions: A function func(user_input, report_func) -> bool taking user input and returning bool.
+                          See `report_func` parameter for report_func specification.
+
+        :param report_func: Function for printing errors and warnings from conditions.
+        :type report_func: Function func(message)  -- taking one argument message.
+        """
+        super().__init__()
+        self._title = title
+        self._value = value
+        self._conditions = conditions
+        self._report_func = report_func
+
+        self._user_prompt = _("Enter a new value for '%(title)s' and press %(enter)s") % {
+                                # TRANSLATORS: 'title' as a title of the entry
+                                "title": title,
+                                # TRANSLATORS: 'enter' as the key ENTER
+                                "enter": Prompt.ENTER
+                               }
+        self._no_separator = False
+
+    @property
+    def title(self):
+        """Title of the item we want to get user input for.
+
+        :returns: Name of the item.
+        :rtype: str.
+        """
+        return self._title
+
+    @title.setter
+    def title(self, title):
+        """Set title of the item we want to get user input for.
+
+        :param title: Item title.
+        :type title: str.
+        """
+        self._title = title
+
+    @property
+    def value(self):
+        """Value of the item we want to set.
+
+        Optional.
+
+        :returns: Value user set or was predefined.
+        :rtype: str. Default: None
+        """
+        return self._value
+
+    @value.setter
+    def value(self, value):
+        """Set value of the item we want to set.
+
+        Optional.
+
+        :param value: Value of the item.
+        :type value: str.
+        """
+        self._value = value
+
+    @property
+    def no_separator(self):
+        """Print separator or hide user input as a actual screen part?
+
+        :returns: False if separator should be printed. True if not.
+        :rtype: bool. Default: False
+        """
+        return self._no_separator
+
+    @no_separator.setter
+    def no_separator(self, no_separator):
+        """Should the separator be printed?
+
+        :param no_separator: True to disable separator, False otherwise.
+        :type no_separator: bool.
+        """
+        self._no_separator = no_separator
+
+    def widget_name(self):
+        """Get name with value for the widget."""
+        msg = self.title
+
+        if self._value:
+            msg += "\n"
+            msg += self._value
+
+        return msg
+
+    def get_user_input(self):
```

Comment:
I would like to replace `get_user_input` and `get_user_input_callback` with a method `run(*args, **kwargs)`. The method shouldn't return anything, so the only way how to get the method is with the property `value`.

Instead of:
```
w = TextWidget(self._nfs_server_input.widget_name())
self._container.add(w, self._nfs_server_input.get_user_input_callback)
```


We could do:
`self._container.add(self._nfs_server_dialog.widget, self._nfs_server_dialog.run)`

Reply:
I really don't like the ``*args, **kwargs`` methods but I like the idea. 
The `return` is not mandatory and I guess we can go with ``*args, **kwargs`` with information that the parameters will be ignored.